### DISCUSSION TITLE
Add transient write mode for floppy images

### DISCUSSION THREAD

**Discussion by chungy** [86Box/86Box#3414] [Ideas]
**Is your feature request related to a problem? Please describe.**

Some software installers will write out registration information to the first disk of their setup, and will often fail if the disk is write-protected. In the interest of keeping pristine images (or even to allow for a comparison before and after), I currently have to copy files to a temporary location and use that image in a full write mode.

It would be excellent if 86box could have a "transient writable" mode, where any changes made to the floppy image are stored in-memory only and discarded as soon as the floppy is removed from the 86box machine.

**Describe the solution you'd like**

A new menu entry for floppy disks (maybe MO and Zip disks? Perhaps not so necessary on them, though), call it "Existing image (in-memory writes)" or something of the sort.

**Comment by cold-brewed** [comment]
Sounds very much like [snapshot mode](https://qemu.readthedocs.io/en/latest/system/images.html#snapshot-mode) in qemu via `-snapshot`:

> If you use the option -snapshot, all disk images are considered as read only. When sectors in written [sic], they are written in a temporary file created in /tmp.

The writes are then discarded when you close the system. It's a handy feature.

However, the name "snapshot mode" is confusing because it is unrelated to VM snapshots.

**Comment by chungy** [comment]
Indeed it is that idea, likewise VirtualBox's "immutable" mode for disks.

**Comment by ArnoldUK** [comment]
Why dont you simply make the floppy image read only on the host system if it is not to be written to. That'd be the equivalent of sliding the read only switch on the floppy disk.

**Comment by jguillaumes** [comment]
If you do that the installer will fail.

**Comment by ArnoldUK** [comment]
I'm not quite sure if you are using real floppy disks or images created with winimage or some other imaging software ? 

EDIT: Sorry I thought you were talking about physical disk images. Yes, the transient mode for floppy soft images would be a good feature but using a carbon copy of the image would work just as well.

I still don't know how you would compare them if the disk image is transient write mode and stored in memory ? You would still have to commit the memory to the disk image and compare it to the original disk on the host.

**Comment by OBattler** [comment]
Eventually, there should be 4 modes in my opinion:
- Read-only (already implemented);
- Transient write;
- Write back at emulator close;
- Write back at sector write end (already implemented - the current write mode).
